WinUIGallery/App.xaml.cs

```diff
@@ -167,87 +166,196 @@ private async void EnsureWindow(IActivatedEventArgs args = null)
 
             ThemeHelper.Initialize();
 
+            if (args.PreviousExecutionState == ApplicationExecutionState.Terminated
+                    || args.PreviousExecutionState == ApplicationExecutionState.Suspended)
+            {
+                try
+                {
+                    await SuspensionManager.RestoreAsync();
+                }
+                catch (SuspensionManagerException)
+                {
+                    //Something went wrong restoring state.
+                    //Assume there is no state and continue
+                }
+
+                Window.Current.Activate();
+
+                UpdateNavigationBasedOnSelectedPage(rootFrame);
+                return;
+            }
+
             Type targetPageType = typeof(NewControlsPage);
             string targetPageArguments = string.Empty;
 
-            if (args != null)
+            if (args.Kind == ActivationKind.Launch)
             {
-                if (args.Kind == ActivationKind.Launch)
+                targetPageArguments = ((Windows.ApplicationModel.Activation.LaunchActivatedEventArgs)args).Arguments;
+            }
```

Comment:
It seems that in my testing, the Kind was always set to ActivationKind.Launch which is definitely not correct when using the protocol. I think there might be issues in the framework that prevent this from working properly.